index.js

```
@@ -0,0 +1,60 @@
+//index.js
+'use strict';
+
+//dependencies
+var express = require('express');
+var AWS = require('aws-sdk');
+var validator = require('./validate');
+var bodyParser = require('body-parser');
+var compression = require('compression');
+
+// initialize variables
+var app = express();
+AWS.config.update({region: 'us-west-2'});
+var s3 = new AWS.S3({apiVersion: '2006-03-01'});
+
+app.use(compression());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json({limit: '50mb'}));
+
+
+function testHeader(headers){
+  var testCache = (headers['cache-control'] == 'no-cache');
+  var contentType = (headers['content-type'] == 'application/json');
+  var encoding = headers['accept-encoding'].split(", ").includes('gzip');
+  return testCache && contentType && encoding
+}
+
+// test app
+app.get('/start', function (req, res) {
+  res.send('Start successfull!');
+});
+
+
+// send validated json to s3
+app.post('/push', function(req, res) {
+  var jsonObject = req.body;
+  if(validator.isValid(jsonObject) && testHeader(req.headers)){
+   //set params for S3 entry
+   var uploadParams = {Bucket: '', Key: '', Body: ''};
+   uploadParams.Bucket = process.env.BUCKET_NAME;
+   uploadParams.Key = jsonObject["mapname"] + '.json';
+   console.log("Key of S3 object: " + uploadParams.Key);
+   uploadParams.Body = JSON.stringify(jsonObject);
+   s3.upload (uploadParams, function (err, data) {
+    if (err) {
+      console.log("Error", err);
+    }
+    if (data) {
+      console.log("Upload Success", data.Location);
+    }
+   });
+  }else{
+    console.log('Json is not valid.');
```

please return the error of the validator instead. Can be done later.